src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDK.scala

```diff
@@ -118,7 +122,100 @@ class TextAnalyticsLanguageDetection(override val textAnalyticsOptions: Option[T
     }
 }
 
+object TextSentimentV4 extends ComplexParamsReadable[TextSentimentV4]
+class TextSentimentV4(override val textAnalyticsOptions: Option[TextAnalyticsRequestOptions] = None,
+                      override val uid: String = randomUID("TextSentimentV4"))
+  extends TextAnalyticsSDKBase[SentimentScoredDocumentV4](textAnalyticsOptions)
+    with HasConfidenceScoreCol {
+  logClass()
+
+  override def outputSchema: StructType = SentimentResponseV4.schema
+
+  override protected val invokeTextAnalytics: String => TAResponseV4[SentimentScoredDocumentV4]= (text: String) =>{
+    val textSentimentResultCollection = textAnalyticsClient.analyzeSentimentBatch(
+      Seq(text).asJava, "en", textAnalyticsOptions.orNull)
+
+    def getConfidenceScore(score: SentimentConfidenceScores): SentimentConfidenceScoreV4 = {
+      SentimentConfidenceScoreV4(
+        score.getNegative,
+        score.getNeutral,
+        score.getPositive)
+    }
+    def getTarget(target: TargetSentiment): TargetV4 = {
+      TargetV4(
+        target.getText,
+        target.getSentiment.toString,
+        getConfidenceScore(target.getConfidenceScores),
+        target.getOffset,
+        target.getLength)
+    }
+    def getAssessment(assess: AssessmentSentiment): AssessmentV4 = {
+      AssessmentV4(
+        assess.getText,
+        assess.getSentiment.toString,
+        getConfidenceScore(assess.getConfidenceScores),
+        assess.isNegated,
+        assess.getOffset,
+        assess.getLength)
+    }
+
+    def getSentenceSentiment(sentencesent: SentenceSentiment): SentimentSentenceV4 = {
+      SentimentSentenceV4(
+        sentencesent.getText,
+        sentencesent.getSentiment.toString,
+        getConfidenceScore(sentencesent.getConfidenceScores),
+        if(sentencesent.getOpinions == null){

```

Comment:
A cleaner way to do this is 
```scala
Option(sentencesent.getOpinions).map(...)
```
`Option(some_val)` returns `None` if `some_val` is null, and return `Some(some_val)` otherwise, and you can chain maps from there.